src/arrayMethodSort.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFunction ? compareFunction(this[j], this[j + 1])
+          > 0 : String(this[j]) > String(this[j + 1])) {
+          const TEMP = this[j];
```

Comment:
instead of this logic you can just add the default value for the compareFunction